### PR TITLE
Update libcudacxx to 1.8

### LINKED ISSUE #251: [FEA] Upgrade libcudacxx to 1.8.0
**Is your feature request related to a problem? Please describe.**
Can we please update the default libcudacxx version to 1.8.0 thus we can use the new `cuda::std::array` feature?

**Describe the solution you'd like**
Update the libcudacxx to 1.8.

**Additional context**
@robertmaynard mentioned that 1.8.0 introduced CMake support so it requires more changes to avoid conflicting targets being constructed

### PR DESCRIPTION
**Pull request by robertmaynard**
Fixes #251

### CODE REVIEW

File: rapids-cmake/cpm/libcudacxx.cmake [hunk 1]
```diff
@@ -51,97 +51,65 @@ Result Variables
...
+    # `install/include`, while libcudacxx tries to mark it as an user include, since rmm uses
+    # CMake's default of system include. Compilers when provided the same include as both user and
+    # system always goes with system.
+
+    # Now while rmm could also mark `install/include` as system this just pushes the issue to
+    # another dependency which isn't built by RAPIDS and comes by and marks `install/include` as
+    # system.
+
+    # Instead the more reliable option is to make sure that we get libcudacxx to be placed in an
+    # unique include path that to other project will use. In the case of rapids-cmake we install the
```
Comment (bdice) [inline rapids-cmake/cpm/libcudacxx.cmake:95]:
```suggestion
    # unique include path that the other project will use. In the case of rapids-cmake we install the
```

File: rapids-cmake/cpm/libcudacxx.cmake [hunk 2]
```diff
@@ -51,97 +51,65 @@ Result Variables
...
+    # headers to `include/rapids/thrust`
```
Comment (bdice) [inline rapids-cmake/cpm/libcudacxx.cmake:96]:
```suggestion
    # headers to `include/rapids/libcudacxx`
```

File: testing/cpm/cpm_libcudacxx-export.cmake [hunk 3]
```diff
@@ -21,6 +21,12 @@ rapids_cpm_init()
 rapids_cpm_libcudacxx(BUILD_EXPORT_SET frank INSTALL_EXPORT_SET test)
 rapids_cpm_libcudacxx(INSTALL_EXPORT_SET test2)
 
+add_library(example INTERFACE )
```
Comment (bdice) [inline testing/cpm/cpm_libcudacxx-export.cmake:24]:
```suggestion
add_library(example INTERFACE)
```

**Review by bdice — APPROVED** [review]
Looks like CI passed on https://github.com/rapidsai/cudf/pull/11596. I have a few minor suggestions.

File: testing/cpm/cpm_libcudacxx-export.cmake (hunk 3 above)
Comment (robertmaynard) [reply testing/cpm/cpm_libcudacxx-export.cmake:24]: I believe that we can't change this as it will fail the style checks

Comment (bdice) [reply testing/cpm/cpm_libcudacxx-export.cmake:24]: Hmm, I didn’t expect that. Is the whitespace meaningful in the syntax?

Comment (robertmaynard) [reply testing/cpm/cpm_libcudacxx-export.cmake:24]: No. But cmake-format likes to remove white space before ( and )

**Pull request merged**